R/backend-hana.R

```diff
@@ -57,7 +57,8 @@ sql_translation.HDB <- function(con) {
 
 # nocov start
 #' @export
-db_table_temporary.HDB <- function(con, table, temporary) {
+db_table_temporary.HDB <- function(con, table, temporary, ...) {
+  check_dots_empty0(...)
```

Comment:
FWIW I think it's technically incorrect to check for empty dots in an S3 method because it means that a subclass that calls a parent method with `NextMethod()` can't add any additional arguments, somewhat defeating the point of using `...` in the first place.

That's why we have `check_dots_used()`, which must be called from the generic.

Reply:
I now remember there was a discussion about `check_dots_used()` vs `check_dots_empty()`, e.g. https://github.com/tidyverse/dplyr/issues/6599.
But the reason to use `check_dots_empty()` instead of `check_dots_used()` was mostly due to tidyselect. I guess for most generics in this PR it wouldn't make sense to add a tidyselect argument. So should I switch to `check_dots_used()` in the generic instead?

Reply:
Yeah, I think that’s a bit safer.